vochain/scrutinizer/process.go

```diff
@@ -114,24 +134,41 @@ func (s *Scrutinizer) ProcessList(entityID []byte, namespace uint32,
 }
 
 // ProcessCount return the number of processes indexed
-func (s *Scrutinizer) ProcessCount() int64 {
-	c, err := s.db.Count(&Process{}, nil)
+func (s *Scrutinizer) ProcessCount(entityID []byte) int64 {
+	var c int
+	var err error
+	if len(entityID) == 0 {
+		// If no entity ID, return the cached count of all processes
+		return atomic.LoadInt64(s.countTotalProcesses)
+	} else {
```

Comment:
you can drop this else and unindent the Count, as the "if" branch always returns. then you don't need the `var` lines above either.